crev-data/src/proof/content.rs

```diff
@@ -0,0 +1,128 @@
+use crate::Url;
+use chrono::{self, prelude::*};
+use failure::bail;
+
+use self::super::review;
+use self::super::trust::TrustDraft;
+use crate::proof;
+use crate::proof::{Proof, ProofType};
+use crate::Result;
+
+pub type Date = chrono::DateTime<FixedOffset>;
+
+pub trait ContentCommon: std::fmt::Debug + std::string::ToString + ContentCommonClone {
+    fn as_any(&self) -> &dyn std::any::Any;
+
+    fn date(&self) -> &Date;
+    fn set_date(&mut self, date: &Date);
+    fn date_utc(&self) -> chrono::DateTime<Utc> {
+        self.date().with_timezone(&Utc)
+    }
+
+    fn author(&self) -> &crate::PubId;
+    fn set_author(&mut self, id: &crate::PubId);
+
+    fn author_id(&self) -> crate::Id {
+        self.author().id.clone()
+    }
+
+    fn author_url(&self) -> Url {
+        self.author().url.clone()
+    }
+
+    fn draft_title(&self) -> String;
+    fn validate_data(&self) -> Result<()>;
+
+    fn parse(s: &str) -> Result<Self>
+    where
+        Self: Sized;
+
+    fn parse_draft(&self, s: &str) -> Result<Self>
+    where
+        Self: Sized;
+
+    fn proof_type(&self) -> ProofType;
```

Comment:
If we are to be consistent, then there shouldn't be any `ProofType`.  `fn proof_type_name(&self) -> &'static str` should be enough.